sdk/Pulumi/Provider/Provider.cs

```diff
@@ -600,6 +602,41 @@ internal static Microsoft.Extensions.Hosting.IHost BuildHost(
                 })
                 .Build();
         }
+
+        private static string? GetEngineAddress(string[] args)
+        {
+            var cleanArgs = new List<string>();
+
+            for (int i = 0; i < args.Length; i++)
+            {
+                var arg = args[i];
+
+                // Skip logging-related arguments
+                if (arg == "--logtostderr") continue;
+                if (arg.StartsWith("-v")) continue;
```

Comment:
`StartsWith` does a culture-sensitive comparison. It would be better to use `arg.StartsWith("-v", StringComparison.Ordinal)` here.

We should enable some FxCop rules (or equivalent) to catch things like this.

Reply:
https://github.com/pulumi/pulumi-dotnet/pull/559